xdsl/interpreters/riscv.py

```diff
@@ -281,9 +282,9 @@ def get_data(module_op: ModuleOp) -> dict[str, RawPtr]:
                                 assert (
                                     False
                                 ), f"Unexpected directive {data_op.directive.data}"
-                    return data
-        else:
+        if data is None:
```

Comment:
```suggestion
       assert data is not None, "Could not find data section"
```

Reply:
I'm a bit confused by all the `if` checks and `assert` statements.

Reply:
They're markers of me being lazy, really they should be raising better errors, I'd rather either change them to errors now, or keep the weird if data is None structure.

Reply:
Swapped asserts